neqo-transport/src/connection/mod.rs

```diff
@@ -2909,7 +2918,7 @@ impl Connection {
         let val = self
```

Comment:
Just above is one place where `data: impl AsRef<u8>` is probably necessary.  You have the `.as_ref()` call here, but that's a noop here without the change.